### PR TITLE
Added methods and test-cases for Assets V3 endpoints (excluding WAS)

### PR DESCRIPTION
**Pull request by MitaliUpadhyay**
# Description

In this PR I have added scripts and test-cases for :
- Explore -> Assets -> Cloud Resources V3 endpoints. 
- Explore -> Assets -> Hosts V3 endpoints. 
- Explore -> Assets -> All V3 endpoints. 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

To test code-changes of this PR, I have run testcases of `tests/io/test_assets_api_v3.py` . All testcases of this file are working fine and getting passed. 

**Test Configuration**:
* Python Version(s) Tested: 3.9

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tenable/io/v3/explore/assets/api.py [hunk 1]
```diff
@@ -106,3 +103,237 @@ def search_webapp(self,
...
+                rows in text/csv format for each call with row headers.
+        Returns:
+            Iterable:
+                The iterable that handles the pagination for the job.
+            requests.Response:
+                If ``return_resp`` is set to ``True``, then a response
+                object is returned instead of an iterable.
+
+        Examples:
+            >>> tio.v3.assets.search_webapp(filter=('netbios_name', 'eq', 'SCCM'),
```
Comment (alpesh-te) [inline tenable/io/v3/explore/assets/api.py:174]: Set the correct method example

File: tenable/io/v3/explore/assets/api.py [hunk 2]
```diff
@@ -106,3 +103,237 @@ def search_webapp(self,
...
```
Comment (alpesh-te) [inline tenable/io/v3/explore/assets/api.py:252]: Set the correct example

File: tenable/io/v3/explore/assets/api.py [hunk 3]
```diff
@@ -106,3 +103,237 @@ def search_webapp(self,
...
```
Comment (alpesh-te) [inline tenable/io/v3/explore/assets/api.py:330]: You should set example with multiple type

File: tests/io/test_assets_api_v3.py [hunk 4]
```diff
@@ -12,27 +13,136 @@ def test_explore_assets_v3_search_was_assets(api):
...
+
+@pytest.mark.vcr()
+def test_explore_assets_v3_search_all_assets(api):
+    '''
+    Test the search all the assets endpoint
+    '''
+    operating_system_list = ['linux*']
+    source_list = ['AWS']
+    sort = [('last_observed', 'desc')]
+    filters = ('and', ('operating_systems', 'wc', operating_system_list), ('sources', 'eq', source_list))
```
Comment (alpesh-te) [inline tests/io/test_assets_api_v3.py:121]: I would suggest you should add filter of multiple type ['webapp', 'host'] etc

**Review by alpesh-te — APPROVED** [review]
Need to addressed minor fixes

**Review by varap11 — APPROVED** [review]

**Pull request merged**